contracts/TransferLimitModule.sol

```diff
@@ -185,7 +192,7 @@ contract TransferLimitModule is Module, SignatureDecoder, SecuredTokenTransfer {
 
         // We transfer the calculated tx costs to the tx.origin to avoid sending it to intermediate contracts that have made calls
         if (gasPrice > 0) {
-            handlePayment(startGas, gasLimit, gasPrice, gasToken, refundReceiver);
+            handlePayment(gasLimit, gasPrice, gasToken, refundReceiver);
```

Comment:
In this approach, why do we need `gasLimit` and `gasPrice` and not just `refund` (`gasLimit * gasPrice` can be calculated off-chain)

We can leave it like this for now, but the idea why we don't fail the whole transaction if the inner transaction fails is that the relay service will else pay for failing transactions. And these do happen even if we estimate before (e.g. investing into an ico -> this is sometimes time bound).

Reply:
@rmeissner good points. Merged `gasLimit` and `gasPrice` to `refund`.

The module now doesn't revert on `execTransactionFromModule`. It updates spent values after transfer success, and on failure emits an event. Even on failure relayer is refunded if there's enough of `gasToken` in safe.

One edge case I can think of is if a token contract returns false on failing transfers, instead of reverting. When this happens, `execTransactionFromModule` still returns true.